lib/SILGen/SILGenFunction.cpp

```diff
@@ -612,6 +612,19 @@ void SILGenFunction::emitGeneratorFunction(SILDeclRef function, Expr *value) {
   RegularLocation Loc(value);
   Loc.markAutoGenerated();
 
+  // Default argument generators of function typed values return noescape
+  // functions. Strip the escape to noescape function conversion.
+  if (function.kind == SILDeclRef::Kind::DefaultArgGenerator) {
```

Comment:
Since this whole 'if' block only has any effect when assertions are on, maybe wrap it inside an #ifndef NDEBUG?

Reply:
``value`` is set to ``conv->getSubExpr()``